lib/simbricks/pic/if.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 Max Planck Institute for Software Systems, and
+ * National University of Singapore
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "lib/simbricks/pic/if.h"
+
+void SimbricksPicIfDefaultParams(struct SimbricksBaseIfParams *params) {
+  SimbricksBaseIfDefaultParams(params);
+  params->upper_layer_proto = SIMBRICKS_PROTO_ID_PIC;
+  // fit DMA writes with size 8192
```

Comment:
can you explain this hard cded 8192 ?  @sidharth-sundar

Reply:
I'm not sure on that, it's a remnant from the memory interface (lib/simbricks/mem/if.c). It's setting the amount of space required in shared memory, which is used for the input and output message queues. I think it's allocating enough space to fit an 8192B DMA write alongside the remainder of the packet body, but currently we don't issue such large writes.